server/service/src/invoice/inbound_return/generate_lines.rs

```diff
@@ -0,0 +1,269 @@
+use crate::{service_provider::ServiceContext, ListError, ListResult};
+use repository::{
+    EqualFilter, InvoiceLineFilter, InvoiceLineRepository, ItemRow, RepositoryError, StockLineRow,
+};
+use util::uuid::uuid;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct InboundReturnLine {
+    pub id: String,
+    pub reason_id: Option<String>,
+    pub note: Option<String>,
+    pub number_of_packs: f64,
+    pub stock_line_row: StockLineRow,
+    pub item_row: ItemRow,
+    pub packs_issued: Option<f64>,
+}
+
+pub struct ExistingLinesInput {
+    pub item_id: String,
+    pub return_id: String,
+}
+
+pub struct GenerateInboundReturnLinesInput {
+    pub outbound_shipment_line_ids: Vec<String>,
+    pub existing_lines_input: Option<ExistingLinesInput>,
+}
+
+pub fn generate_inbound_return_lines(
+    ctx: &ServiceContext,
+    _store_id: &str,
+    GenerateInboundReturnLinesInput {
+        outbound_shipment_line_ids,
+        existing_lines_input: include_existing_lines,
+    }: GenerateInboundReturnLinesInput,
+) -> Result<ListResult<InboundReturnLine>, ListError> {
+    let new_return_lines = if !outbound_shipment_line_ids.is_empty() {
+        generate_new_return_lines(ctx, outbound_shipment_line_ids)?
+    } else {
+        vec![]
+    };
+
+    let existing_return_lines =
+        if let Some(ExistingLinesInput { item_id, return_id }) = include_existing_lines {
+            get_existing_return_lines(ctx, &item_id, &return_id)?
+        } else {
+            vec![]
+        };
+
+    let return_lines: Vec<InboundReturnLine> = existing_return_lines
+        .into_iter()
+        .chain(new_return_lines.into_iter())
```

Comment:
I've seen chain, extend and append in different places - I'm not confident on when to pick one over the other?